Makefile

```diff
@@ -27,99 +28,93 @@ endif
 	DC_EXEC=$(DOCKER_COMPOSE) exec -e USE_FOUNDRY_BUNDLE=${USE_FOUNDRY_BUNDLE} -e DATABASE_URL=${MYSQL_URL} -e MONGO_URL=${MONGO_URL}
 else
-	CI='true'
-	DOCKER_COMPOSE=$(DOCKER_COMPOSE_BIN) -f docker-compose.yaml -f docker-compose.ci.yaml
 	# CI needs to be ran in no-tty mode
 	DC_EXEC=$(DOCKER_COMPOSE) exec -e USE_FOUNDRY_BUNDLE=${USE_FOUNDRY_BUNDLE} -e DATABASE_URL=${MYSQL_URL} -e MONGO_URL=${MONGO_URL} -T
 endif
 
-DOCKER_PHP=${DC_EXEC} php
+PHP=php${PHP_VERSION}
+DOCKER_PHP=${DC_EXEC} ${PHP}
 
-# From inside the containers, docker host ip is different in linux and macos
-UNAME_S := $(shell uname -s)
-ifeq ($(UNAME_S),Linux)
-    XDEBUG_HOST=172.17.0.1
-endif
-ifeq ($(UNAME_S),Darwin)
-    XDEBUG_HOST=host.docker.internal
-endif
+ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+$(eval $(RUN_ARGS):;@:)
 
 help:
 	@fgrep -h "###" $(MAKEFILE_LIST) | fgrep -v fgrep | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
 validate: fixcs sca test database-validate-mapping ### Run fixcs, sca, full test suite and validate migrations
 
-test: docker-start vendor ### Run PHPUnit tests suite
-	@$(eval filter ?= '.')
-	@${DC_EXEC} -e USE_ORM=${USE_ORM} -e USE_ODM=${USE_ODM} php vendor/bin/simple-phpunit --configuration ${PHPUNIT_CONFIG_FILE} --filter=$(filter)
+test: vendor ### Run PHPUnit tests suite
+	@${DC_EXEC} -e USE_ORM=${USE_ORM} -e USE_ODM=${USE_ODM} ${PHP} vendor/bin/simple-phpunit --configuration ${PHPUNIT_CONFIG_FILE} $(ARGS)
 
-fixcs: docker-start bin/tools/cs-fixer/vendor ### Run PHP-CS-Fixer
-	@${DOCKER_PHP} bin/tools/cs-fixer/vendor/friendsofphp/php-cs-fixer/php-cs-fixer --no-interaction --diff -v fix
+fixcs: bin/tools/cs-fixer/vendor ### Run PHP-CS-Fixer
+	@${DC_EXEC} -e PHP_CS_FIXER_IGNORE_ENV=1 ${PHP} php -d 'xdebug.mode=off' bin/tools/cs-fixer/vendor/friendsofphp/php-cs-fixer/php-cs-fixer --no-interaction --diff -v fix
 
-bin/tools/cs-fixer/vendor: vendor bin/tools/cs-fixer/composer.json bin/tools/cs-fixer/composer.lock
-	@${DOCKER_PHP} composer bin cs-fixer install
+bin/tools/cs-fixer/vendor: bin/tools/cs-fixer/composer.json bin/tools/cs-fixer/composer.lock
+	@$(MAKE) --no-print-directory docker-start
+	@$(MAKE) --no-print-directory vendor
+	@$(MAKE) --no-print-directory composer bin cs-fixer install
 
-sca: docker-start bin/tools/phpstan/vendor ### Run static analysis
-	@${DOCKER_PHP} bin/tools/phpstan/vendor/phpstan/phpstan/phpstan analyse
+sca: bin/tools/phpstan/vendor ### Run static analysis
+	@${DOCKER_PHP} php -d 'xdebug.mode=off' bin/tools/phpstan/vendor/phpstan/phpstan/phpstan analyse
 
-bin/tools/phpstan/vendor: vendor bin/tools/phpstan/composer.json $(wildcard bin/tools/phpstan/composer.lock)
-	@${DOCKER_PHP} composer bin phpstan install
+bin/tools/phpstan/vendor: bin/tools/phpstan/composer.json bin/tools/phpstan/composer.lock
+	@$(MAKE) --no-print-directory docker-start
+	@$(MAKE) --no-print-directory vendor
+	@$(MAKE) --no-print-directory composer bin phpstan install
```

Comment:
- removing `docker-start` and `vendor` as requirement of the make target + calling them _inside_ the target improve `make sca` and `make fixcs` which no more call `composer install` everytime
- I've manage to disable xdebug for composer / phpstan / php cs fixer targets